scipy/_lib/_array_api.py

```diff
@@ -283,13 +283,20 @@ def xp_assert_equal(actual, desired, check_namespace=True, check_dtype=True,
     return np.testing.assert_array_equal(actual, desired, err_msg=err_msg)
 
 
-def xp_assert_close(actual, desired, rtol=1e-07, atol=0, check_namespace=True,
+def xp_assert_close(actual, desired, rtol=None, atol=0, check_namespace=True,
                     check_dtype=True, check_shape=True, err_msg='', xp=None):
     __tracebackhide__ = True  # Hide traceback for py.test
     if xp is None:
         xp = array_namespace(actual)
     desired = _strict_check(actual, desired, xp, check_namespace=check_namespace,
                             check_dtype=check_dtype, check_shape=check_shape)
+
+    floating = xp.isdtype(desired.dtype, ('real floating', 'complex floating'))
+    if rtol is None and floating:
+        rtol = xp.finfo(desired.dtype).eps**0.5
```

Comment:
If we do this:
```suggestion
        rtol = xp.finfo(desired.dtype).eps**0.5 * 4
```
Then the default `rtol` for float64 is `5.960464477539063e-08`, close enough to `1e-7` for all existing tests to pass without adjustment.  The downside is that it sets the default `rtol` for `float32` to `0.0013810679320049757`, which seems a little loose.